.github/workflows/build-and-test-ipc.yaml

```diff
@@ -51,36 +51,46 @@ jobs:
       NANOARROW_ARROW_TESTING_DIR: '${{ github.workspace }}/arrow-testing'
 
     steps:
-      - name: Checkout repo
-        uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
```

Comment:
Yeah I don't see you doing anything that would require a full checkout (e.g. changing branch/using tags etc.) :+1: